sklearn/feature_selection/from_model.py

```diff
@@ -44,59 +109,18 @@ def transform(self, X, threshold=None):
         X_r : array of shape [n_samples, n_selected_features]
             The input samples with only the selected features.
         """
-        check_is_fitted(self, ('coef_', 'feature_importances_'), 
+        check_is_fitted(self, ('coef_', 'feature_importances_'),
                         all_or_any=any)
 
         X = check_array(X, 'csc')
-        # Retrieve importance vector
-        if hasattr(self, "feature_importances_"):
-            importances = self.feature_importances_
-
-        elif hasattr(self, "coef_"):
-            if self.coef_ is None:
-                msg = "This model is not fitted yet. Please call fit() first" 
-                raise NotFittedError(msg)
-
-            if self.coef_.ndim == 1:
-                importances = np.abs(self.coef_)
-            else:
-                importances = np.sum(np.abs(self.coef_), axis=0)
-
+        importances = _get_feature_importances(self)
```

Comment:
Is this correct? Shouldn't it be called on `self.estimator` or `self.estimator_`?